### PR TITLE
telegram-desktop: 4.8.1 -> 4.8.3

### PR DESCRIPTION
**Pull request by NickCao**
###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes (or backporting 23.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/networking/instant-messengers/telegram/telegram-desktop/default.nix [hunk 1]
```diff
@@ -73,15 +75,15 @@ let
...
+  version = "4.8.3";
   # Note: Update via pkgs/applications/networking/instant-messengers/telegram/tdesktop/update.py
 
   src = fetchFromGitHub {
     owner = "telegramdesktop";
     repo = "tdesktop";
     rev = "v${version}";
     fetchSubmodules = true;
-    sha256 = "0mxxfh70dffkrq76nky3pwrk10s1q4ahxx2ddb58dz8igq6pl4zi";
+    sha256 = "1xpmkv8lix3iifn96cx3fcd8nv58jpygl406l9ay47fia543vy3y";
```
Comment (drupol) [inline pkgs/applications/networking/instant-messengers/telegram/telegram-desktop/default.nix:86]: How about using `hash` next time?

**Review by drupol — APPROVED** [review]

File: pkgs/applications/networking/instant-messengers/telegram/telegram-desktop/default.nix (hunk 1 above)
Comment (NickCao) [reply pkgs/applications/networking/instant-messengers/telegram/telegram-desktop/default.nix:86]: We can do it this time.

**Review by drupol — APPROVED** [review]
Good for me after eval is done!

File: pkgs/applications/networking/instant-messengers/telegram/telegram-desktop/default.nix [hunk 2]
```diff
@@ -73,15 +75,15 @@ let
...
+  version = "4.8.3";
   # Note: Update via pkgs/applications/networking/instant-messengers/telegram/tdesktop/update.py
 
   src = fetchFromGitHub {
     owner = "telegramdesktop";
     repo = "tdesktop";
     rev = "v${version}";
     fetchSubmodules = true;
-    sha256 = "0mxxfh70dffkrq76nky3pwrk10s1q4ahxx2ddb58dz8igq6pl4zi";
+    hash = "1xpmkv8lix3iifn96cx3fcd8nv58jpygl406l9ay47fia543vy3y";
```
Comment (drupol) [inline pkgs/applications/networking/instant-messengers/telegram/telegram-desktop/default.nix:86]: Indeed, the hash is invalid here and needs to be recalculated.

**Review by drupol — CHANGES_REQUESTED** [review]

**Review by drupol — APPROVED** [review]

**Pull request merged**